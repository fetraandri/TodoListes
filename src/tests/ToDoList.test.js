import React from "react";
import { create } from "react-test-renderer";
import ToDoList from "../App";
import { act } from "react-dom/test-utils"; 
import renderer from "react-test-renderer"


describe("ToDoList component", () => {
  test("it matches the snapshot", () => {
    const component = create(<ToDoList />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('ajout d"un tache dans la nouvelle liste', () => {
    const tasks = [
      { id: 1, name: 'Task 1', done: false },
      { id: 2, name: 'Task 2', done: false },
    ];
    const component = renderer.create(<ToDoList tasks={tasks} />);

   

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("qui va verifier la tache", () => {
    const component = create(<ToDoList />);
    const instance = component.getInstance();

    instance.handleCheck({ target: { checked: true, value: "Learn Jest" } });
    expect(instance.state.checkedTasks).toContain("Learn Jest");
  });

  it('should match the snapshot', () => {
    const task = { id: 1, name: 'Test task', done: false };
    const tree = renderer
      .create(<task task={task} />)
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});
