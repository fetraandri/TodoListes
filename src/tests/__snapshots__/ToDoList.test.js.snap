// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList component ajout d"un tache dans la nouvelle liste 1`] = `
<div
  className="full"
>
  <div
    className="todo"
  >
    <form
      onSubmit={[Function]}
    >
      <h1>
        to do:
      </h1>
      <input
        onChange={[Function]}
        placeholder="Nouvelle tâche"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Ajouter
      </button>
    </form>
    <div />
  </div>
  <div
    className="done"
  >
    <h1>
      Done: 
    </h1>
    <ul />
  </div>
</div>
`;

exports[`ToDoList component it matches the snapshot 1`] = `
<div
  className="full"
>
  <div
    className="todo"
  >
    <form
      onSubmit={[Function]}
    >
      <h1>
        to do:
      </h1>
      <input
        onChange={[Function]}
        placeholder="Nouvelle tâche"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Ajouter
      </button>
    </form>
    <div />
  </div>
  <div
    className="done"
  >
    <h1>
      Done: 
    </h1>
    <ul />
  </div>
</div>
`;

exports[`ToDoList component should match the snapshot 1`] = `
<task
  task={
    Object {
      "done": false,
      "id": 1,
      "name": "Test task",
    }
  }
/>
`;
